osu.Game/Overlays/Rankings/Tables/RankingsTable.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Extensions.IEnumerableExtensions;
+
+namespace osu.Game.Overlays.Rankings.Tables
+{
+    public abstract class RankingsTable<TModel> : TableContainer
+    {
+        protected const int TEXT_SIZE = 14;
+        private const float horizontal_inset = 20;
+        private const float row_height = 25;
+        private const int items_per_page = 50;
+
+        private readonly int page;
+        private readonly FillFlowContainer backgroundFlow;
+
+        public IReadOnlyList<TModel> Rankings
+        {
+            set
+            {
+                Content = null;
+                backgroundFlow.Clear();
+
+                if (value?.Any() != true)
+                    return;
+
+                value.ForEach(_ => backgroundFlow.Add(new TableRowBackground()));
+
+                Columns = CreateHeaders();
+                Content = value.Select((s, i) => CreateContent(page * items_per_page - (items_per_page - i), s)).ToArray().ToRectangular();
+            }
+        }
+
+        protected RankingsTable(int page)
```

Comment:
do we ever start on a page that isn't the first page?

Reply:
Can you elaborate on this one? Still unanswered.

Reply:
We need that variable only to calculate indexes on the left side of the table.

Reply:
yes, but is it required in the ctor?

Reply:
Umm, for me yes. Do you have any suggestions about this may be?

Reply:
So you will be reconstructing the table each page? Why is the `Rankings` property even a thing then? If you're only going to be populating it once ever, why not also put that in the ctor?